cmd/zq/zq.go

```diff
@@ -75,29 +77,38 @@ func init() {
 }
 
 type Command struct {
-	zctx         *resolver.Context
-	ifmt         string
-	ofmt         string
-	dir          string
-	path         string
-	jsonTypePath string
-	outputFile   string
-	verbose      bool
-	stats        bool
-	quiet        bool
-	showVersion  bool
+	zctx           *resolver.Context
+	ifmt           string
+	ofmt           string
+	dir            string
+	path           string
+	jsonTypePath   string
+	jsonInferPath  bool
+	jsonPathRegexp string
+	outputFile     string
+	verbose        bool
+	stats          bool
+	quiet          bool
+	showVersion    bool
 	zio.Flags
 }
 
 func New(f *flag.FlagSet) (charm.Command, error) {
 	cwd, _ := os.Getwd()
 	c := &Command{zctx: resolver.NewContext()}
+
+	// x509.14:00:00-15:00:00.log.gz (open source zeek)
+	// x509_20191101_14:00:00-15:00:00+0000.log.gz (corelight)
+	c.jsonPathRegexp = `([a-zA-Z0-9_]+)(?:\.|_\d{8}_)\d\d:\d\d:\d\d\-\d\d:\d\d:\d\d(?:[+\-]\d{4})?\.log(?:$|\.gz)`
+
 	f.StringVar(&c.ifmt, "i", "auto", "format of input data [auto,bzng,ndjson,zeek,zjson,zng]")
 	f.StringVar(&c.ofmt, "f", "zng", "format for output data [bzng,ndjson,table,text,types,zeek,zjson,zng]")
 	f.StringVar(&c.path, "p", cwd, "path for input")
 	f.StringVar(&c.dir, "d", "", "directory for output data files")
 	f.StringVar(&c.outputFile, "o", "", "write data to output file")
 	f.StringVar(&c.jsonTypePath, "j", "", "path to json types file")
+	f.BoolVar(&c.jsonInferPath, "inferpath", false, "infer zeek _path from file name (for ndjson objects without _path)")
```

Comment:
I'm not the end user, so take my input here with a large dalop of salt, but... I would just assume get rid of the `jsonInferPath` option and if a json file does not have appear to have `_path` in its records it tries to get them using the filename.

Reply:
That sounds reasonable... @philrz what do you say?

Reply:
It has been a while since I've paged in my strong opinions about JSON typing, but, yes, that sounds reasonable to me too.

Reply:
@philrz 👍 this addresses my question above (https://github.com/brimsec/zq/pull/455#issuecomment-603265371) too.

Reply:
(done)